src/HalfIntegers.jl

```diff
@@ -521,6 +521,16 @@ struct Half{T<:Integer} <: HalfInteger
 end
 Half{T}(x::Real) where T<:Integer = half(Half{T}, twice(T,x))
 Half{T}(x::Half{T}) where T<:Integer = x
+function Half{T}(x::Rational) where T<:Integer
+    if x.den == one(x.den)
+        tx = twice(T, x.num)
+    elseif x.den == one(x.den) + one(x.den)
```

Comment:
```suggestion
    if x.den == 1
        tx = twice(T, x.num)
    elseif x.den == 2
```
`one(::BigInt)` allocates, so this will speed up the `Half{T}(::Rational{BigInt})` case even more.

Reply:
Indeed, good catch.